### ISSUE TITLE
`max-len`: ignore commentaries after string in case "ignoreStrings": true

### ISSUE THREAD

**Issue by andirsun** [eslint-stylistic/eslint-stylistic#912]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

This is my rule configuration:
```
'@stylistic/max-len': [
        'error',
        {
          code: 120,
          ignoreStrings: true,
          ignoreTemplateLiterals: true,
          ignoreUrls: true,
          ignoreRegExpLiterals: true,
        },
```

This line fails as expected.
```
// And this is a very long commentary that should trigger the lint rule as it has more than 120 characters lorem ipsum lorem ipsum lorem ipsum
const myString = 'This is my string'
```

But When I put the commentary in the same line is not triggering the expected error.
```
const myString = 'This is my string' // And this is a very long commentary that should trigger the lint rule as it has more than 120 characters lorem ipsum lorem ipsum lorem ipsum
```

### Reproduction
https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHF1b3RlczogW1wiZXJyb3JcIiwgXCJkb3VibGVcIl0gKi9cbmNvbnN0IGEgPSBcImJcIjsgLy8gVGhpcyBpcyBhIHZlcnkgbG9uZyBjb21tZW50YXJ5IHRoYXQgc2hvdWxkIHRocm93IGEgZXNsaW50IGFsZXJ0IGFzIGl0IGhhcyBtb3JlIHRoYW4gMTIwIGNoYXJhY3RlcnMgbG9yZW0gaXBzdW0gbG9yZW0gaXBzdW1cblxuLy8gVGhpcyBpcyBhIHZlcnkgbG9uZyBjb21tZW50YXJ5IHRoYXQgc2hvdWxkIHRocm93IGEgZXNsaW50IGFsZXJ0IGFzIGl0IGhhcyBtb3JlIHRoYW4gMTIwIGNoYXJhY3RlcnMgbG9yZW0gaXBzdW0gbG9yZW0gaXBzdW1cbmNvbnN0IGIgPSBcImJcIjsgXG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im1heC1sZW4iOlsiZXJyb3IiLHsiY29kZSI6MTIwLCJpZ25vcmVDb21tZW50cyI6ZmFsc2UsImlnbm9yZVN0cmluZ3MiOnRydWUsImlnbm9yZVRyYWlsaW5nQ29tbWVudHMiOmZhbHNlLCJpZ25vcmVUZW1wbGF0ZUxpdGVyYWxzIjp0cnVlLCJpZ25vcmVVcmxzIjp0cnVlLCJpZ25vcmVSZWdFeHBMaXRlcmFscyI6dHJ1ZX1dfSwibGFuZ3VhZ2VPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Hi 👋,

You might be looking for either:
- [ignoreComments](https://eslint.style/rules/max-len#ignorecomments) (skips all comments)
- [ignoreTrailingComments](https://eslint.style/rules/max-len#ignoretrailingcomments) (skips only end-of-line comments)

instead of `ignoreStrings`. Would either of these solve your use case?

**Comment by andirsun** [comment]
@9romise Hey, in this case is not working, even if I remove `ignoreStrings` line, the rule is not throwing an error. (I need it to throw it)

<img width="755" height="128" alt="Image" src="https://github.com/user-attachments/assets/ebf2f4d6-cd8f-4ed6-b106-69c196284fd1" />
<img width="1218" height="94" alt="Image" src="https://github.com/user-attachments/assets/d769e85f-3bc3-49cb-806e-99b12d8d6d97" />

**Comment by 9romise** [comment]
Sorry, I misunderstood earlier.
The `ignoreStrings` behavior intentionally skips entire lines containing strings rather than trimming string lengths.
This is the historical design inherited from ESLint core, as documented in https://github.com/eslint/eslint/issues/5805.
